FDB not compiling on macOS Catalina

```/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_SYSTEM_NO_DEPRECATED -DNO_INTELLISENSE -DUSE_UCONTEXT -I/Users/mpilman/Projects/fdb/foundation -I. -I/Users/mpilman/Projects/fdb/foundation/flow -Iflow -I/Users/mpilman/Projects/fdb/foundation/flow/libeio -Iflow/libeio -IboostProject-prefix/src/boostProject -IfmtProject-prefix/include -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk   -ggdb -fno-omit-frame-pointer -maes -mmmx -mavx -msse4.2 -std=c++17 -stdlib=libc++ -Wno-unknown-warning-option -Wno-dangling-else -Wno-sign-compare -Wno-comment -Wno-unknown-pragmas -Wno-delete-non-virtual-dtor -Wno-format -Werror -Wno-error=format -Wno-deprecated -fvisibility=hidden -Wreturn-type -Wsequence-point -fPIC -MD -MT flow/CMakeFiles/flow.dir/Platform.cpp.o -MF flow/CMakeFiles/flow.dir/Platform.cpp.o.d -o flow/CMakeFiles/flow.dir/Platform.cpp.o -c /Users/mpilman/Projects/fdb/foundation/flow/Platform.cpp
In file included from /Users/mpilman/Projects/fdb/foundation/flow/Platform.cpp:8:
/Users/mpilman/Projects/fdb/foundation/flow/Platform.h:618:1: error: static declaration of 'aligned_alloc' follows non-static declaration
aligned_alloc(size_t alignment, size_t size)
^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/malloc/_malloc.h:50:10: note: previous declaration is here
void    *aligned_alloc(size_t __alignment, size_t __size) __result_use_check __alloc_size(2) __OSX_AVAILABLE(10.15) __IOS_AVAILABLE(13.0) __TVOS_AVAILABLE(13.0) __WATCHOS_AVAILABLE(6.0);
         ^
1 error generated.```

It seems that FoundationDB does not compile on MacOS Catalina as this version of MacOS introduced a function for `aligned_alloc`.

--- a/cmake/ConfigureCompiler.cmake
+++ b/cmake/ConfigureCompiler.cmake
@@ -221,8 +221,13 @@ else()
   # Check whether we can use dtrace probes
   include(CheckSymbolExists)
   check_symbol_exists(DTRACE_PROBE sys/sdt.h SUPPORT_DTRACE)
+  check_symbol_exists(aligned_alloc stdlib.h HAS_ALIGNED_ALLOC)
+  message(STATUS "Has aligned_alloc: ${HAS_ALIGNED_ALLOC}")
   if(SUPPORT_DTRACE)
     add_compile_definitions(DTRACE_PROBES)
+  endif()
+  if(HAS_ALIGNED_ALLOC)
+    add_compile_definitions(HAS_ALIGNED_ALLOC)
   endif()
 
   if(CMAKE_COMPILER_IS_GNUCXX)

--- a/flow/Platform.h
+++ b/flow/Platform.h
@@ -524,6 +524,7 @@ inline static void aligned_free(void* ptr) { free(ptr); }
 inline static void* aligned_alloc(size_t alignment, size_t size) { return memalign(alignment, size); }
 #endif
 #elif defined(__APPLE__)
+#if !defined(HAS_ALIGNED_ALLOC)
 #include <cstdlib>
 inline static void* aligned_alloc(size_t alignment, size_t size) {
 	// Linux's aligned_alloc() requires alignment to be a power of 2.  While posix_memalign()
@@ -540,6 +541,7 @@ inline static void* aligned_alloc(size_t alignment, size_t size) {
 	posix_memalign(&ptr, alignment, size);
 	return ptr;
 }
+#endif
 inline static void aligned_free(void* ptr) { free(ptr); }
 #endif
 

